packages/noya-state/src/reducers/history.ts

```diff
@@ -0,0 +1,99 @@
+import produce from 'immer';
+import { ApplicationState, Action, reducer } from './application';
+import { createInitialInteractionState } from './interaction';
+
+export type HistoryEntry = {
+  actionType: Action[0];
+  timestamp: number;
+  state: ApplicationState;
```

Comment:
Can `state` just be a `SketchFile`? Is there anything else we actually want to preserve when we undo/redo (for now)?

Reply:
I think so, should we store selection state similar to other design tools?

Reply:
Leaving this for now per our Slack convo.